tests/stats/test_median.py

```diff
@@ -6,42 +6,175 @@
 """Tests for Drift package."""
 import numpy as np
 import pytest
+import pandas as pd
 
+from typing import Tuple
 from nannyml.datasets import load_synthetic_car_loan_dataset
 from nannyml.stats import SummaryStatsMedianCalculator
 
-# @pytest.fixture(scope="module")
-# def status_sum_result() -> Result:
-#     reference, analysis, _ = load_synthetic_car_loan_dataset()
 
-#     calc = SummaryStatsStdCalculator(
-#         column_names=[
-#             'car_value',
-#             'debt_to_income_ratio',
-#             'driver_tenure'
-#         ],
-#     ).fit(reference)
-#     return calc.calculate(data=analysis)
+@pytest.fixture
+def binary_classification_data() -> Tuple[pd.DataFrame, pd.DataFrame]:  # noqa: D103
+    reference, monitored, _ = load_synthetic_car_loan_dataset()
+    return reference.head(15_000), monitored.tail(5_000)
 
 
-def test_stats_median_calculator_with_default_params_should_not_fail():  # noqa: D103
-    reference, analysis, _ = load_synthetic_car_loan_dataset()
+def test_stats_median_calculator_with_default_params_should_not_fail(  # noqa: D103
+    binary_classification_data
+):
+    reference, monitored = binary_classification_data
     try:
         calc = SummaryStatsMedianCalculator(
             column_names=['car_value', 'debt_to_income_ratio', 'driver_tenure'],
         ).fit(reference)
-        _ = calc.calculate(data=analysis)
+        _ = calc.calculate(data=monitored)
     except Exception:
         pytest.fail()
 
 
-def test_stats_median_calculator_should_not_fail_given_nan_values():  # noqa: D103
-    reference, analysis, _ = load_synthetic_car_loan_dataset()
-    reference.loc[20000:30000, 'car_value'] = np.NaN
+def test_stats_median_calculator_should_not_fail_given_nan_values(  # noqa: D103
+    binary_classification_data
+):
+    reference, monitored = binary_classification_data
+    reference.loc[1000:11000, 'car_value'] = np.NaN
     try:
         calc = SummaryStatsMedianCalculator(
             column_names=['car_value'],
         ).fit(reference)
-        _ = calc.calculate(data=analysis)
+        _ = calc.calculate(data=monitored)
     except Exception:
         pytest.fail()
+
+
+def test_stats_median_calculator_results(binary_classification_data):  # noqa: D103
```

Comment:
This style is quite verbose. Might be better to split into multiple tests (since you've made it very "splittable") with the results as a fixture (so they're only calculated once, check `realized_performance_metrics` fixture in `test_regression`) or just assert the on the full dataframe. 


Ideally each test covers a single assertion, so you know instantly what is impacted when the test fails. I do admit that a lot of the other "result tests" are not structured like that. This just breaks the previous structure